### PR TITLE
[WIP] e2e test framework proposal changes

### PR DESCRIPTION
**Pull request by jsturtevant**
**What this PR does / why we need it**:
Proposed changes for #855.  

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by jsturtevant** [comment]
This currently requires a few more changes but wanted to get thoughts on structure before making the final effort.

File: test/e2e/azure_test.go [hunk 1]
```diff
@@ -68,53 +68,69 @@ var _ = Describe("Workload cluster creation", func() {
...
+					},
+					CNIManifestPath:              e2eConfig.GetVariable(CNIPath),
+					WaitForClusterIntervals:      e2eConfig.GetIntervals(specName, "wait-cluster"),
+					WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
+					WaitForMachineDeployments:    e2eConfig.GetIntervals(specName, "wait-worker-nodes"),
+				})
 			})
 
-			Context("Creating a accessible load balancer", func() {
+			It("Creating a accessible load balancer", func() {
```
Comment (CecileRobertMichon) [inline test/e2e/azure_test.go:133]:
```suggestion
			It("Creates an accessible load balancer", func() {
```

is this more readable / grammatically correct?

File: test/e2e/azure_test.go [hunk 2]
```diff
@@ -125,7 +141,7 @@ var _ = Describe("Workload cluster creation", func() {
 				})
 			})
 
-			Context("Validating network policies", func() {
+			It("Validating network policies", func() {
```
Comment (cpanato) [inline test/e2e/azure_test.go:144]: then the same here:

```suggestion
			It("Validates the network policies", func() {
```

File: test/e2e/azure_test.go (hunk 1 above)
Comment (jsturtevant) [reply test/e2e/azure_test.go:133]: good catch.  now that it is closer to a common ginko structure the `It` clauses the tests will read better with updates like this. Thanks!

**Comment by jsturtevant** [comment]
This is not going to work as is. I've put some notes in https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/855#issuecomment-669413143.  Going to close this PR until we have a bit more direction

**Pull request closed without merging**